stream-chat-android-core/src/main/java/io/getstream/chat/android/client/utils/Result.kt

```diff
@@ -17,7 +17,7 @@ public data class Result<T : Any> @Deprecated(
     public constructor(error: ChatError) : this(null, error)
 
     val isSuccess: Boolean
-        get() = data != null
+        get() = data != null && error == null
```

Comment:
This should never be needed. Instead of making this change, let's find a way to eliminate the [last remaining use of the deprecated `Result` constructor](https://github.com/GetStream/stream-chat-android/blob/9e2b1ef20be75ad250ee502bbbe712c8c5f94947/stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/controller/ChannelControllerImpl.kt#L707). Then we can remove the deprecation, make the constructor private, and we'll be sure that either `data` or `error` are set in all cases.